cmd/virt-chroot/tap-device-maker.go

```diff
@@ -4,26 +4,26 @@ import (
 	"fmt"
 	"strconv"
 
-	"github.com/songgao/water"
 	"github.com/spf13/cobra"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
 )
 
-func createTapDevice(name string, owner uint, group uint, isMultiqueue bool) error {
+func createTapDevice(name string, owner uint, group uint, queueNumber int) error {
 	var err error = nil
-	config := water.Config{
-		DeviceType: water.TAP,
-		PlatformSpecificParams: water.PlatformSpecificParams{
-			Name:    name,
-			Persist: true,
-			Permissions: &water.DevicePermissions{
-				Owner: owner,
-				Group: group,
-			},
-			MultiQueue: isMultiqueue,
-		},
+	tapDevice := &netlink.Tuntap{
+		LinkAttrs:  netlink.LinkAttrs{Name: name},
+		Mode:       unix.IFF_TAP,
+		NonPersist: false,
+		Queues:     queueNumber,
```

Comment:
I guess it is also only needed to determine if the multiqueu flag is set on the interface, right?

Reply:
You're right, as can be seen in https://github.com/vishvananda/netlink/blob/337442361b22376df1edb7fc665b1830c4ece2eb/link_linux.go#L1089 .

This also means that we could be a bit more conservative in our code, not change the api, and just use a queue number > 0 whenever we're on multi-queue.

Your thoughts on this ?

Reply:
I just looked a little bit closer there. It looks like it also opens as many file descriptors as the queue count and does not close it. That is probably not much of an issue, since they get closed when virt-chroot exits.

I think I would feel slightly better if we provide it the right number of queues, since we would pass in what the api wants us to pass in.

Reply:
Darn, you're right!

Sure thing; I'll move the `converter` code to a common place and use that.